webapp.js

```diff
@@ -495,9 +496,56 @@
             connection.addRequestCallback(this.onRequest.bind(this,stream,connection))
             connection.tryRead()
         },
+        render400HTML: function(request) {
+            var handler = new WSC.BaseHandler(request)
+            handler.app = this
+            handler.request = request
+            if (request.method == "HEAD") {
+                handler.responseLength = 0
+                handler.writeHeaders(400)
+                handler.finish()
+                return
+            }
+            if (this.opts.optCustom400) {
+                this.fs.getByPath(this.opts.optCustom400location, (file) => {
+                if (! file.error) {
+                    file.file( function(filee) {
+                    var reader = new FileReader();
+                    reader.onload = function(e){
+                        this.useDefaultMime = false
```

Comment:
This is pretty weird to switch on and off a global variable like this. If I was still actively maintaining this project I would ask you to do something less weird here :-)

Reply:
Yeah I think the same thing. It was the easiest way I could think of the mime type not rendering as the requested file extension.